oligo_designer_toolsuite/oligo_specificity_filter/_filter_seed_region.py

```diff
@@ -0,0 +1,196 @@
+        """Apply bowtie filter to all batches in parallel"""
+        # generater the seed region coordinates
+        oligo_DB = self.seed_region_creation.apply(oligo_DB)
+
+        # Some bowtie initializations, change the names
+        index_exists = False
+        index_name = os.path.basename(file_reference_DB)
+        # Check if bowtie index exists
+        for file in os.listdir(self.dir_seed_region):
+            if re.search(f"^{index_name}.*", file):
+                index_exists = True
+                break
+        # Create bowtie index if none exists
+        if not index_exists:
+            command1 = "bowtie-build " + file_reference_DB + " " + index_name
+            process = Popen(command1, shell=True, cwd=self.dir_seed_region).wait()
+
+        oligo_DB_seed = self._extract_seed_regions(oligo_DB)
+        genes = list(oligo_DB_seed.keys())
+        filtered_oligo_DBs = Parallel(n_jobs=n_jobs)(
+            delayed(self._run_bowtie_seed_region)(
+                oligo_DB_seed[gene], oligo_DB[gene], gene, index_name
+            )
+            for gene in genes
+        )
+
+        # reconstruct the oligos_DB
+        for gene, filtered_oligo_DB in zip(genes, filtered_oligo_DBs):
+            oligo_DB[gene] = filtered_oligo_DB
+
+        return oligo_DB
+
+    def _run_bowtie_seed_region(self, gene_DB_seed, gene_DB, gene, index_name):
+        """Run Bowtie alignment tool to find regions of local similarity between sequences, where sequences are probes and transcripts.
+        Bowtie identifies all allignments between the probes and transcripts and returns the number of mismatches and mismatch position for each alignment.
+
+        :return: DataFrame with processed bowtie alignment search results.
+        :rtype: pandas.DataFrame
+        """
+
+        file_probe_fasta_gene = self._create_fasta_file(
+            gene_DB_seed, self.dir_fasta, gene
+        )
+        file_bowtie_gene = os.path.join(
+            self.dir_seed_region,
+            f"bowtie_{gene}.txt",
+        )
+        command = (
+            "bowtie -x "
+            + index_name
+            + " -f -a -v "
+            + str(self.num_mismatches)
+            + " "
+            + file_probe_fasta_gene
+            + " "
+            + file_bowtie_gene
+        )
+
+        process = Popen(command, shell=True, cwd=self.dir_seed_region).wait()
+
+        # read the results of the bowtie search
+        bowtie_results = self._read_bowtie_output(file_bowtie_gene)
+        # filter the DB based on the bowtie results
+        matching_probes = self._find_matching_probes(bowtie_results)
+        filtered_gene_DB = self._filter_matching_probes(gene_DB, matching_probes)
+        # remove the temporary files
+        os.remove(os.path.join(self.dir_seed_region, file_bowtie_gene))
+        os.remove(os.path.join(self.dir_fasta, file_probe_fasta_gene))
+        return filtered_gene_DB
+
+    def _read_bowtie_output(self, file_bowtie_gene):
```

Comment:
Is this the same as _read_bowtie_output in the Bowtie filter? If so, create a base class with a common function